elastic/kibana
# [Reporting] Restore legacy compatibility shim for PDF job creation

Resolves https://github.com/elastic/kibana/issues/108118

This PR reverts a breaking change Reporting had in the master branch, but not 7.x. This is a low-risk PR: when it is backported to 7.x it will largely be clean-up since logic already exists there, but uses JS not TypeScript.

7.x versions have a "compatibility shim" that gives support for POST URLs that were copied in Kibana 6.2 and below. This shim was removed from the master branch in fef8485 as a way to ease the TypeScript conversion of export types and improve maintainability. New initiatives emphasize a more careful approach to removing support for deprecated features. Before removing  this support we will collect telemetry to understand the impact to users.

This PR restores the behavior from 7.x, which is to fully support a URL that was copied in Kibana 6.2.4 or below.

## Testing
1. Start a cluster with Elasticsearch and Kibana version **6.2.4**
2. Import data and create a visualization. Save the visualization and copy the POST URL from the Reporting menu
    ![image](https://user-images.githubusercontent.com/908371/129273147-759b5a5f-b860-4c72-8d25-5201eba40cf8.png)
3. Confirm that the POST URL has a `savedObjectId` parameter, and not a `relativeUrls` parameter
4. Upgrade the cluster to **7.14.0**.
   - Note that the POST URL copied in 6.2.4 still works
5. Upgrading to 8.0 can be tricky, because the existence of system indices that were created in 6.2.4 will not allow ES 8.0 to run. To ease testing, I did the following:
    1. Export all of the Kibana saved objects
    2. Start a new ES using the latest 8.0-snapshot, and Kibana running this branch of Kibana.
    3. Import the original test data and the saved objects that were exported in 7.14.
7. Confirm that the POST URL copied in 6.2.4 still works.
8. Confirm that the job listing shows an appropriate warning:
    ![image](https://user-images.githubusercontent.com/908371/129298630-4b219fed-dff7-4340-82dd-52e194411222.png)

**Resources:**
Docker-compose config: https://gist.github.com/tsullivan/07ce488527265e9ddbdb5f0e7f5df13e
Test data to import: https://gist.github.com/tsullivan/1f999db105499e553aee020ab21c1e9b
Example saved objects: https://gist.github.com/tsullivan/9247922158b077c449415bc429ff5e7e
Script to fire post url: https://gist.github.com/tsullivan/0b86d5e1165f6460a876b4f63012fdd0
